iroha-cli/interactive/impl/interactive_transaction_cli.cpp

```diff
@@ -0,0 +1,443 @@
+                        "2. Send to Iroha (send)", "3. Add command (add)",
+                        "0. Go Back and start new tx (b)"};
+
+      result_handlers_["1"] = &InteractiveTransactionCli::parseSaveFile;
+      result_handlers_["save"] = &InteractiveTransactionCli::parseSaveFile;
+
+      result_handlers_["2"] = &InteractiveTransactionCli::parseSendToIroha;
+      result_handlers_["send"] = &InteractiveTransactionCli::parseSendToIroha;
+
+      result_handlers_["3"] = &InteractiveTransactionCli::parseAddCommand;
+      result_handlers_["add"] = &InteractiveTransactionCli::parseAddCommand;
+
+      result_handlers_["0"] = &InteractiveTransactionCli::parseGoBack;
+      result_handlers_["b"] = &InteractiveTransactionCli::parseGoBack;
+    }
+
+    InteractiveTransactionCli::InteractiveTransactionCli(
+        std::string creator_account) {
+      creator_ = creator_account;
+      // TODO: synchronize counter with iroha
+      tx_counter_ = 0;
+      assign_command_handlers();
+      assign_result_handlers();
+    }
+
+    void InteractiveTransactionCli::run() {
+      std::string line;
+      bool is_parsing = true;
+      current_context_ = MAIN;
+      printMenu("Forming a new transactions, choose command to add: ",
+                commands_points_);
+      // Creating a new transaction, increment local tx_counter
+      ++tx_counter_;
+      while (is_parsing) {
+        line = promtString("> ");
+        switch (current_context_) {
+          case MAIN:
+            is_parsing = parseCommand(line);
+            break;
+          case RESULT:
+            is_parsing = not parseResult(line);
+            break;
+        }
+      }
+    }
+
+    bool InteractiveTransactionCli::parseCommand(std::string line) {
+      std::transform(line.begin(), line.end(), line.begin(), ::tolower);
+      // Find in main handler map
+      auto command = parser::split(line)[0];
+      if (command == "b" || command == "0") {
+        // Return back to main menu
+        // Stop parsing
+        return false;
+      }
+      auto it = command_handlers_.find(command);
+      if (it != command_handlers_.end()) {
+        auto res = (this->*it->second)(line);
+        if (res) {
+          commands_.push_back(res);
+          current_context_ = RESULT;
+          printMenu("Command is formed. Choose what to do:", result_points_);
+        }
+      } else {
+        std::cout << "Command not found" << std::endl;
+      }
+      // Continue parsing
+      return true;
+    }
+
+    std::shared_ptr<iroha::model::Command>
+    InteractiveTransactionCli::parseAddAssetQuantity(std::string line) {
+      std::vector<std::string> notes = {"Account Id to add assets", "Asset id",
+                                        "Amount to add (integer part)",
+                                        "Amount to add (fractional part)"};
+
+      auto params = parseParams(line, ADD_ASSET_QTY, notes);
+      if (not params.has_value()) {
```

Comment:
With object presence need to check size of vector, because direct access useing below.

Reply:
Also, may be provide reason for user about error.